independent-projects/arc/tcks/cdi-tck-porting-pkg/src/main/java/io/quarkus/arc/tck/porting/ContextsImpl.java

```diff
@@ -14,13 +14,18 @@
 import io.quarkus.arc.ManagedContext;
 
 public class ContextsImpl implements Contexts<Context> {
+
+    private InjectableContext.ContextState savedState;
```

Comment:
Is it guaranteed that `setInactive()`/`setActive()` is called from the same thread? I would at least make the field `volatile`.

Reply:
There is only a few usages in TCK (about 8?) some of which are CDI Full too and in none of them I saw multiple threads being used. But we can surely roll with `volatile` as well.
This is mostly a workaround for what I think should be a TCK challenge I have yet to create.

Reply:
It is very crude indeed :-D
Since I aimed to raise a TCK challenge to begin with, I just went for simplest possible fix. Not to mention, for something like CDI Lite, you can only really toggle on/off req. context and the class already work with `ManagedContext` which will only work for req. context in Arc.
Nonetheless, you are right to point that out; let's do the map approach instead.